In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from lead_scoring.src.data_loader import charger_donnees, apercu


In [2]:
df= charger_donnees()
apercu(df)

Dimensions : 11162 lignes x 17 colonnes

Valeurs manquantes :
age          0
job          0
marital      0
education    0
default      0
balance      0
housing      0
loan         0
contact      0
day          0
month        0
duration     0
campaign     0
pdays        0
previous     0
poutcome     0
deposit      0
dtype: int64

Distribution cible :
deposit
0    5873
1    5289
Name: count, dtype: int64

Taux de souscription : 47.38%


In [3]:

df.describe()

,age,balance,day,duration,campaign,pdays,previous,deposit
count,11162.000000,11162.000000,11162.000000,11162.000000,11162.000000,11162.000000,11162.000000,11162.000000
mean,41.231948,1528.538524,15.658036,371.993818,2.508421,51.330407,0.832557,0.473840
std,11.913369,3225.413326,8.420740,347.128386,2.722077,108.758282,2.292007,0.499338
min,18.000000,-6847.000000,1.000000,2.000000,1.000000,-1.000000,0.000000,0.000000
25%,32.000000,122.000000,8.000000,138.000000,1.000000,-1.000000,0.000000,0.000000
50%,39.000000,550.000000,15.000000,255.000000,2.000000,-1.000000,0.000000,0.000000
75%,49.000000,1708.000000,22.000000,496.000000,3.000000,20.750000,1.000000,1.000000
max,95.000000,81204.000000,31.000000,3881.000000,63.000000,854.000000,58.000000,1.000000


## Analyse — Statistiques descriptives

- **`balance`** : forte dispersion (std = 3225, min = -6847€) → présence d'outliers, à traiter en preprocessing
- **`duration`** : très étalée → variable potentiellement prédictive mais biaisée (inconnue avant l'appel)
- **`pdays`** : min à -1 → valeur codée "jamais contacté", pas une vraie valeur numérique → à traiter en S2
- **`campaign`** : max à 63 contacts → valeur aberrante
- **`deposit`** : moyenne à 0.47 → dataset quasi-équilibré, pas besoin de rééchantillonnage
